omod/src/test/java/org/openmrs/module/idgen/web/controller/AutoGenerationOptionControllerTest.java

```diff
@@ -0,0 +1,152 @@
+
+import org.apache.commons.beanutils.PropertyUtils;
+import org.apache.poi.hssf.record.formula.functions.True;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.PatientIdentifierType;
+import org.openmrs.module.idgen.AutoGenerationOption;
+import org.openmrs.module.idgen.service.IdentifierSourceService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.test.Util;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.idgen.web.RestTestConstants;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+
+/**
+ * Tests CRUD operations for {@link AutoGenerationOption}s via web service calls
+ */
+public class AutoGenerationOptionControllerTest extends MainResourceControllerTest {
+  
+  private IdentifierSourceService service;
+
+  public PatientIdentifierType ptid;
+
+  @Before
+	public void before() throws Exception {
+    executeDataSet("org/openmrs/module/idgen/include/TestData.xml");
+    this.service = Context.getService(IdentifierSourceService.class);
+  }
+
+  @Override
+	public String getURI() {
+		return "idgen/autogenerationoption";
+  }
+  
+  @Override
+	public String getUuid() {
+		return "1";
+  }
+
+  @Override
+	public long getAllCount() {
+		return service.getAutoGenerationOptions(ptid).size();
+	}
+  
+  
+  @Test
+  public void shouldGetAnAutoGenerationOptionById() throws Exception {
+    MockHttpServletRequest req = request(RequestMethod.GET, getURI() + "/" + getUuid());
+    SimpleObject result = deserialize(handle(req));
+    
+    AutoGenerationOption autoGenerationOption = service.getAutoGenerationOption(Integer.parseInt(getUuid()));
+    assertEquals(autoGenerationOption.getId(), PropertyUtils.getProperty(result, "id"));
+    assertEquals(autoGenerationOption.getIdentifierType(), PropertyUtils.getProperty(result, "identifierType"));
+    assertEquals(autoGenerationOption.getLocation(), PropertyUtils.getProperty(result, "location"));
+    assertEquals(autoGenerationOption.getSource(), PropertyUtils.getProperty(result, "source"));
+    assertEquals(autoGenerationOption.isAutomaticGenerationEnabled(), PropertyUtils.getProperty(result, "isAutomaticGenerationEnabled"));
+    assertEquals(autoGenerationOption.isManualEntryEnabled(), PropertyUtils.getProperty(result, "isManualEntryEnabled"));
+  }
+
+  @Test
+  public void shouldGetAnAutoGenerationOptionByPatientIdentifierType() throws Exception {
+    MockHttpServletRequest req = request(RequestMethod.GET, getURI() + "/" + ptid);
```

Comment:
What value does ptid have?

Reply:
It holds a uuid of a `PatientIdentifierType`

Reply:
And can you give example of how the url for a rest call using patient identifier type will look like?

Reply:
Can you show me where ptid is assigned a value?

Reply:
I guess I just declared it on line 42 without assigning a value to it.

Reply:
The rest call using patient identifier type will look something like `ws/rest/v1/idgen/autogenerationoption/{patientIdentifyType}

Reply:
So then of what use is it?

Reply:
We do not handle search parameters that way. That one is reserved for the autogenerationoption identifer.
Can you consult @cnagadya to help you with how we deal with such searches?

Reply:
The test won't run unless a `patientIdentifierType` is provided.

I'll consult Christine as you suggest